src/index.ts

```diff
@@ -52,12 +61,9 @@ export default class Help {
     const subject = getHelpSubject(argv)
 
     let command: Config.Command | undefined
+    let topic: Config.Topic | undefined
     if (subject) {
       command = this.config.findCommand(subject)
-    }
-
-    let topic: Config.Topic | undefined
-    if (subject && !command) {
```

Comment:
Is there any point to find the topic if the command doesn't exist?

Reply:
Agreed, that's the  way it was before. The diff makes it tricky to see the change, but I moved both `topic` and `command` together when there is a `subject`:
https://github.com/oclif/plugin-help/blob/7211fd91b2dd020a58f3659cce35bfee04af0c76/src/index.ts#L63-L68

Does that fix what what you're describing?

Reply:
It just seemed before it was checking the `!command` and now you are not. So there could be a case where the command is undefined but you still find the topic?

Reply:
I've changed this a bit now. I've added a test to track the old routing for the previous methods, and made the dependencies for what is shown happen in lock step. I think this makes it easier to see.

Yes, there is a case where the command is undefined and we still have a topic to show. This case is the last chance to find a match for the `subject` before an error is thrown.